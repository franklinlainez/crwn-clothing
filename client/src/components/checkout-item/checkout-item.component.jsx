import React from "react";
import { connect } from "react-redux";

import {
  addItem,
  removeItem,
  clearItemFromCart,
} from "../../redux/cart/cart.actions";

import {
  CheckoutItemContainer,
  ImageContainer,
  QuantityText,
  Text,
  RemoveButtonContainer,
} from "./checkout-item.styles.jsx";

const CheckoutItem = ({ cartItem, clearItem, addItem, removeItem }) => {
  const { name, imageUrl, price, quantity } = cartItem;
  return (
    <CheckoutItemContainer>
      <ImageContainer>
        <img src={imageUrl} alt="item" />
      </ImageContainer>
      <Text>{name}</Text>
      <QuantityText>
        <div onClick={() => removeItem(cartItem)}>&#10094;</div>
        <Text>{quantity}</Text>
        <div onClick={() => addItem(cartItem)}>&#10095;</div>
      </QuantityText>
      <Text>{price}</Text>
      <RemoveButtonContainer onClick={() => clearItem(cartItem)}>
        &#10005;
      </RemoveButtonContainer>
    </CheckoutItemContainer>
  );
};

const mapDispatchToProps = (dispatch) => ({
  clearItem: (item) => dispatch(clearItemFromCart(item)),
  addItem: (item) => dispatch(addItem(item)),
  removeItem: (item) => dispatch(removeItem(item)),
});

export default connect(null, mapDispatchToProps)(CheckoutItem);
