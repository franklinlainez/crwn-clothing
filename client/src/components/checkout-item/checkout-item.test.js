import { ReactWrapper, mount } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import CheckoutItem from "./checkout-item.component";
import {
  addItem,
  clearItemFromCart,
  removeItem,
} from "../../redux/cart/cart.actions";
import { QuantityText } from "./checkout-item.styles";
import { RemoveButtonContainer } from "./checkout-item.styles";

const mockStore = configureMockStore([]);

describe("Checkout Item", () => {
  /**@type ReactWrapper */
  let wrapper;
  let store = mockStore({});

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <CheckoutItem cartItem={{}} />
      </Provider>
    );
  });

  it("render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("call removeItem when the corresponding button is clicked", () => {
    const leftArrow = 0;

    wrapper.find(QuantityText).find("div").at(leftArrow).simulate("click");
    expect(store.getActions()).toContainEqual(removeItem({}));
  });

  it("call addItem when the corresponding button is clicked", () => {
    const rigthArrow = 1;
    wrapper.find(QuantityText).find("div").at(rigthArrow).simulate("click");
    expect(store.getActions()).toContainEqual(addItem({}));
  });

  it("call clearItem when the corresponding button is clicked", () => {
    wrapper.find(RemoveButtonContainer).simulate("click");
    expect(store.getActions()).toContainEqual(clearItemFromCart({}));
  });
});
