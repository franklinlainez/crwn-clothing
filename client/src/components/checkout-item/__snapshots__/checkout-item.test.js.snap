// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Item render the component 1`] = `ReactWrapper {}`;
